### PR TITLE
javascript exorelay: reconnect if exocom goes down

### PR DESCRIPTION
**Pull request by charlierudolph**
replacement for #145 (I updated cucumber-js separately while trying to figure out what was going on)

### CODE REVIEW

File: javascript/exorelay/src/websocket-connector/websocket-connector.ls [hunk 1]
```diff
@@ -56,20 +62,27 @@ class WebSocketConnector extends EventEmitter
...
+      @connect!
 
 
   _on-socket-open: ~>
     @emit 'online'
 
 
   _on-socket-error: (error) ~>
-    | error.errno is 'EADDRINUSE'   =>  @emit 'error', "port #{@exocom-port} is already in use"
-    | error.errno is 'ECONNREFUSED' =>  wait 1_000, @connect
```
Comment (charlierudolph) [inline javascript/exorelay/src/websocket-connector/websocket-connector.ls]: This line was unnecessary for the current tests to pass. It appears the websocket library we are using keeps retrying on the initial connections even though it emits an error.

File: javascript/exorelay/src/websocket-connector/websocket-connector.ls [hunk 2]
```diff
@@ -24,11 +24,17 @@ class WebSocketConnector extends EventEmitter
...
   # Closes the port that ExoRelay is communicating on
-  close: ->
-    return unless @socket
+  close: (done) ->
+    return done! unless @socket
     debug "no longer connected at 'ws://#{@exocom-host}/#{@exocom-port}'"
-    @socket?.close!
-    @emit 'offline'
+    @should-reconnect-on-socket-closed = no
+    switch @socket.readyState
```
Comment (hugobho) [inline javascript/exorelay/src/websocket-connector/websocket-connector.ls]: `@socket.ready-state`

File: javascript/exorelay/src/websocket-connector/websocket-connector.ls [hunk 3]
```diff
@@ -24,11 +24,17 @@ class WebSocketConnector extends EventEmitter
...
+      | WebSocket.CONNECTING => done!
```
Comment (hugobho) [inline javascript/exorelay/src/websocket-connector/websocket-connector.ls]: We don't need to call anything to terminate the socket here?

Comment (charlierudolph) [inline javascript/exorelay/src/websocket-connector/websocket-connector.ls]: Updated. Can't call close as that would error but using `terminate` works.

File: javascript/exorelay/src/websocket-connector/websocket-connector.ls (hunk 2 above)
Comment (charlierudolph) [inline javascript/exorelay/src/websocket-connector/websocket-connector.ls]: fixed

**Comment by kevgo** [comment]
I think `push` just checks the actual commit, and `pr` checks a different commit that is the current commit merged with `master`.

**Pull request merged**